plugins/inputs/statsd/running_stats.go

```diff
@@ -69,12 +82,36 @@ func (rs *RunningStats) AddValue(v float64) {
 		// Reached limit, choose random index to overwrite in the percentile array
 		rs.perc[rand.Intn(len(rs.perc))] = v
 	}
+
+	if len(rs.med) < rs.MedLimit {
+		rs.med = append(rs.med, v)
+	} else {
+		// Reached limit, slice off first element
+		rs.med = rs.med[1:]
+		rs.med[len(rs.med)] = v
+	}
 }
 
 func (rs *RunningStats) Mean() float64 {
 	return rs.k + rs.ex/float64(rs.n)
 }
 
+func (rs *RunningStats) Median() float64 {
+	count := len(rs.med)
+	// Need to sort for median
+	if !rs.SortedMed {
+		sort.Float64s(rs.med)
+		rs.SortedMed = true
+	}
```

Comment:
To keep the temporal order, we need a copy here and do
```suggestion
	// Need to sort for median
        var values []float64
        values = append(values, rs.med...)
	sort.Float64s(values)
```

Reply:
Of course you need to replace `rs.med` with `values` below...